ui/app/services/api.ts

```diff
@@ -129,6 +136,9 @@ export default class ApiService extends Service {
       { post: this.deleteControlGroupToken },
       { post: this.formatErrorResponse },
     ],
+    fetchApi: (...args: [Request]) => {
+      return waitForPromise(window.fetch(...args));
+    },
```

Comment:
There was an issue with the settled state in tests where it wasn't actually waiting for pending fetch requests. This let's the test waiters know to wait for fetch requests from the api service to resolve before proceeding.